src/.nuget/Microsoft.NET.Sdk.Optimization.ReadyToRun/sdk/Sdk.targets

```diff
@@ -0,0 +1,213 @@
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!-- Remaining Work:
+    1) Inject CreateReadyToRunImages at correct entrypoint in the build. Related to https://github.com/dotnet/sdk/issues/2583
+  -->
+  <PropertyGroup>
+    <_IntermediateReadyToRunDirName Condition=" '$(_IntermediateReadyToRunDirName)' == '' ">ReadyToRun</_IntermediateReadyToRunDirName>
+    <_IntermediateReadyToRunDir Condition=" '$(_IntermediateReadyToRunDir)' == '' ">$(IntermediateOutputPath)$(_IntermediateReadyToRunDirName)</_IntermediateReadyToRunDir>
+    <_IntermediateReadyToRunDir>$([System.IO.Path]::GetFullPath($(_IntermediateReadyToRunDir)))</_IntermediateReadyToRunDir>
+
+    <_ReadyToRunOutputDir>$([System.IO.Path]::Combine($(_IntermediateReadyToRunDir),"Output"))</_ReadyToRunOutputDir>
+  </PropertyGroup>
+
+  <Target Name="CreateReadyToRunImages"
+          BeforeTargets="ComputeFilesToPublish"
+          DependsOnTargets="_CheckForUnsupportedReadyToRunCompilation;
+                            _InitializeReadyToRunCompilerProps;
+                            _ComputeReadyToRunInputsAndOutputs;
+                            _RunReadyToRunCompiler;
+                            _PublishReadyToRunOutputs"
+          Condition="'$(ReadyToRun)' == 'true'">
+  </Target>
+
+  <Target Name="_CheckForUnsupportedReadyToRunCompilation"
+          Condition="'$(ReadyToRun)' == 'true'">
+
+    <!-- TODO: Add this message as a resource string to the NET SDK so it can be localized -->
+    <NETSdkError Condition="'3.0' > '$(_TargetFrameworkVersionWithoutV)'"
+        FormattedText="Creating ReadyToRun images is only supported for .NET Core 3.0 and above" />
+
+    <NETSdkError Condition="'$(RuntimeIdentifier)' ==''"
+                 ResourceName="RuntimeIdentifierWasNotSpecified"/>
+
+    <NETSdkError Condition="'$(TargetFramework)' ==''"
+                 ResourceName="AtLeastOneTargetFrameworkMustBeSpecified"/>
+
+    <!-- Determine whether the host OS platform-architecture supports compilation to the desired RuntimeIdentifier -->
+    <PropertyGroup>
+      <_TargetArchitecture>$(PlatformTarget)</_TargetArchitecture>
+      <_TargetArchitecture Condition="$(RuntimeIdentifier.EndsWith('arm64'))">arm64</_TargetArchitecture>
+
+      <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('windows'))">win</_OSPlatform>
+      <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('osx'))">osx</_OSPlatform>
+      <_OSPlatform Condition="$([MSBuild]::IsOSPlatform('linux'))">linux</_OSPlatform>
+      <_OSArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture)</_OSArchitecture>
+
+      <_UnsupportedReadyToRunCompilation>false</_UnsupportedReadyToRunCompilation>
```

Comment:
It is probably better to be unsupported by default and then detect the cases that are allowed. Otherwise hard to reason about cases that aren't explicitly handled, e.g. freebsd.